### PR TITLE
upgrade ruby version, lockfile

### PR DESCRIPTION
**Pull request by msaroufim**
https://deploy-preview-1816--pytorch-dot-org-preview.netlify.app/edge

### Ruby Version Upgrade:
* Updated the Ruby version from `2.7.8` to `3.3.0` in `.ruby-version` and `Gemfile`. [[1]](diffhunk://#diff-1bdb3279c7718f7037156f83c80eb81c58d37b286e027219e055b76249082264L1-R1) [[2]](diffhunk://#diff-d09ea66f8227784ff4393d88a19836f321c915ae10031d16c93d67e6283ab55fL2-R2) - this kills most of the security warnings we have on this repo

### Documentation Updates:
* Changed code block language identifiers to `bash` in `README.md` and updated the `bundler` version to `2.5.23`. [[1]](diffhunk://#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5L17-R44) [[2]](diffhunk://#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5L56-R61)

EDIT: Discussed this a bit more with @cjyabraham and opted to revert the edge related doc changes

### CODE REVIEW

**Comment by svekars** [comment]
I disagree that we should delete the edge.html page without giving it more thought . If it is decided to be deprecated, it needs to be replaced with a redirect to other relevant page. Also, the whole Edge dropdown with just 1 item won't make a lot of sense.

**Comment by msaroufim** [comment]
I chatted with @ali-khosh offline and he said he'd fine with this change if I can guarantee that the redirects are working

**Review by malfet — APPROVED** [review]

File: edge.html [hunk 1]
```diff
@@ -4,54 +4,5 @@
...
-        <a href="https://pytorch.org/executorch/stable/index.html" class="btn btn-lg with-right-arrow">
-          ExecuTorch Documentation
-        </a>
-      </div>
-    </div>
-  </div>
-</div>
-
-
+redirect_to: /executorch-overview
```
Comment (malfet) [inline edge.html:7]: It's weird that this thing is called .html, while it's clearly not :)

**Review by cjyabraham — CHANGES_REQUESTED** [review]
Looks ok, however, you'll also need to update _includes/mobile_menu.html so that it matches your changes to main_menu.html.

**Comment by jennbly** [comment]
I'd suggest that we not have an option in the middle of the drop down menu without a drop down. Could we instead move to under "Ecosystem" or "Learn"?

**Comment by cjyabraham** [comment]
> > Looks ok, however, you'll also need to update _includes/mobile_menu.html so that it matches your changes to main_menu.html.
> 
> can you take a look again please?

The mobile menu still doesn't match with the desktop menu.

BTW, why not put the "Edge" menu item under the "Learn" menu or some other menu rather than have it standing by itself?

File: _includes/mobile_menu.html [hunk 2]
```diff
@@ -67,9 +67,6 @@
           <a>Edge</a>
         </li>
         <ul class="resources-mobile-menu-items">
-          <li>
```
Comment (msaroufim) [inline _includes/mobile_menu.html:70]: @cjyabraham was this not what you were asking about?

**Review by cjyabraham — APPROVED** [review]

**Comment by ali-khosh** [comment]
> BTW, why not put the "Edge" menu item under the "Learn" menu or some other menu rather than have it standing by itself?
Edge is a key strategic bet and we'd like to keep it more visible.

**Comment by cjyabraham** [comment]
I'm having trouble running the site locally on MacOS after this upgrade. I've upgraded ruby, however, I'm now getting this: https://gist.github.com/cjyabraham/5f12926a985e77acb986bc70f4bd7125

Any advice?

**Pull request merged**